modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeReplacementCheck.java

```diff
@@ -0,0 +1,149 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.source.formatter.check;
+
+import com.liferay.petra.string.CharPool;
+import com.liferay.portal.json.JSONArrayImpl;
+import com.liferay.portal.kernel.json.JSONArray;
+import com.liferay.portal.kernel.json.JSONObject;
+import com.liferay.portal.kernel.json.JSONUtil;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.source.formatter.check.util.JavaSourceUtil;
+
+import java.io.InputStream;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @author Nícolas Moura
+ */
+public class UpgradeReplacementCheck extends BaseFileCheck {
+
+	@Override
+	protected String doProcess(
+			String fileName, String absolutePath, String content)
+		throws Exception {
+
+		String newContent = content;
+
+		JSONArray jsonArray = _getReplacementPatternsJSONArray(
+			"replacement-patterns.json");
+
+		for (int i = 0; i < jsonArray.length(); i++) {
+			JSONObject jsonObject = jsonArray.getJSONObject(i);
+
+			if (!_isValidExtension(fileName, jsonObject)) {
+				continue;
+			}
+
+			boolean replaced = false;
+
+			Pattern pattern = Pattern.compile(jsonObject.getString("regEx"));
```

Comment:
Both are valid ways to reference a `regEx`, I found this way more common. About being `pattern`, well I do not think valid because `pattern` is specifically the object within the code, this is the parameter. But I saw here that inside the Pattern is written as "regex" so I can rename it to leave the pattern that Liferay uses.

Reply:
Please see https://github.com/natocesarrego/liferay-portal/pull/1341#discussion_r1290040539